### ISSUE TITLE
participant

### ISSUE THREAD

**Issue by aparrish** [dariusk/NaNoGenMo#41]
I got a copy of propp and a markov chain algorithm, let's do this

**Comment by aparrish** [comment]
It's strange that people are starting (https://github.com/dariusk/NaNoGenMo/issues/11) with the assumption that a "novel" is either (a) a narrative or (b) a string of characters—or a combination of the two, e.g., a manifestation of a narrative in the form of a string of characters.

Is it true that a novel has to "be" a narrative? I think the answer is plainly no, since there are things that are narratives that are not novels, and novels that consist of things other than narratives, or are only narratives if the term "narrative" is applied liberally or as a post-hoc reading. (Not that I have a problem with applying the term "narrative" liberally, or reading things as "narratives" that were not intended to be such.) What formalisms are there to describe novels that aren't based on the idea that the novel is, first and foremost, a narrative? (Is that even a common way to formalize novels? I guess I don't know anything about how novels are studied, from a formal perspective.) What would it mean to "generate" a novel, starting from the idea that the abstractions involved in the generative algorithm wouldn't model narrative in any way?

Another question: is a novel necessarily a string of characters? The answer to this is also plainly no, since there are graphic novels, and novels that have no words at all (see http://en.wikipedia.org/wiki/Wordless_novel). A novel _usually_ uses words (otherwise a "wordless novel" wouldn't be something remarkable enough to warrant its own category), but it's not clear that it _must_ use words. How would you generate a wordless novel? How would you generate a wordless novel that doesn't have a narrative structure?

To back up a bit, it also seems that a lot of narrative-generating algorithms use a particular strategy, which is to model a narrative and then create sentences that relate that narrative, sort of in the same way a sculptor might create a frame and then attach clay to that frame to create a figure. (No idea if this is how any sculptor actually works, but you get the idea.) In this model, there's a sentence (or some other unit of text) that gets generated for every node in the narrative tree. I'm essentializing here, and speaking from a lack of deep familiarity with the way such a system would work, but this model strikes me as untenable for generating a novel—it implies that text _only_ exists in a novel as a means of making the "narrative" manifest. It seems to me that text in novels serves many different systems (tone, style, cohesion, intertextuality, etc.), many of which are orthogonal to "narrative" and need to be modeled separately.

I'm also rejecting here the idea that "style" or "tone" are lexical properties; they're not something that can be "applied" to a stretch of otherwise "neutral" text. At the very least you'd have to think of them as participating in a system of relationships between text units and the contents of those text units. Or something?

All of this isn't intended to start arguments about what counts as a "novel" and I'm certainly not trying to uphold some platonic "novel" ideal to which we all as authors and coders must hew and strive. I'm just trying to think about unusual (and hopefully fruitful) ways to approach the problem at hand.

**Comment by dariusk** [comment]
Yeah, to clarify, the reason I picked 50k words as the only definition of "novel" relevant to NaNoGenMo is to keep it in line with NoNoWriMo.

I'm hoping to get some kind of narrative forward motion in what I generate, mostly because I feel like it's too easy to just, you know, generate 50k markov'ed words and call it a day, and the output is not very interesting. It's the equivalent of firing up a fractal landscape generator, tweaking a few parameters, and generating a map 100x bigger than the one in GTA V. Is it cool? Yeah, it's cool. Is it what I want to do? Nope.

**Comment by catseye** [comment]
@aparrish Well, the way I see it, the word "novel" just means "new".

I take your point, though, because, after all, the word "novel" _does_ mean "new".

**Comment by aparrish** [comment]
One way to interpret the the task we've set ourselves to here is to create a 50k-word text that is "readable," and yeah, narrative-ness is a time-tested way to achieve that. Still, for my purposes I'm resisting the idea that there's an on-and-off switch between "narrative" and "meaninglessness"—I think it's possible for a text to have content that isn't narrative and isn't typically propositional and still be meaningful and readable in at least a somewhat novel-like way.

I was also trying to obliquely get at the fact that a lot of novels don't take the form of stories but are nonetheless definitely novels. I'm thinking of novels with innovative forms, like say Pale Fire or Dictionary of the Khazars, or novels that have multiple forms in the same text, like House of Leaves. It's not clear to me that you could break any of those books down into a single narrative structure—especially if you were trying to map that structure back to the text in a systematic way. (Doing that would be difficult for _any_ novel, but those three in particular present significant challenges)

I guess what I'm saying is that maybe I'll try making a Pale Fire generator.

**Comment by dariusk** [comment]
Yeah, I am absolutely not interested in just generating a Hero's Journey narrative either. This is why I'm really tickled by @catseye's recent output; you get this Beckettian scene of people just wandering around a space.

It's why my first attempt was essentially just ageing people through their dreams.

While I agree that style/tone cannot simply be applied to a "neutral" text via a function, I do think you can reinforce style and tone that way. Specifically, in my dream-novel, the passage that I pulled from the corpus of the molester's dream logs was enhanced by dropping/adding vowels, repeating short sentences, and random capitalization:

> I was in a daupartment store. A chair was displayed A chair was displayed. I saw that it was
> vibrating I saw that it was vibrating. When I touched it, theu floor began to vibrate too. Theun I
> touched som lit Christmas tree lights and they went out.

The style and tone were obviously set by the fact that the passages all had the same human author, but I think I managed to enhance and tweak the tone by running the text through a simple function.
